pyvista/plotting/widgets.py

```diff
@@ -622,6 +622,107 @@ def callback(normal, loc):
 
         return self.add_mesh(plane_clipped_mesh, **kwargs)
 
+    def add_volume_clipper(
+        self,
+        volume,
+        normal='x',
+        invert=False,
+        widget_color=None,
+        value=0.0,
+        assign_to_axis=None,
+        tubing=False,
+        origin_translation=True,
+        outline_translation=False,
+        implicit=True,
+        normal_rotation=True,
+        interaction_event=_vtk.vtkCommand.EndInteractionEvent,
+        origin=None,
+        **kwargs,
+    ):
+        """Clip a volume using a plane widget.
+
+        Parameters
+        ----------
+        volume : pyvista.plotting.volume.Volume
+            The input rendered volume to clip.
```

Comment:
Woah... this is quite a bit different from the rest of the methods in this module that take an input dataset to plot and manipulate. This method takes a dataset that is already plotted. IMO, we need to make that explicitly clear.

Reply:
Can confirm; I'd have no idea what a `Volume` is.

Reply:
Can confirm; I'd have no idea what a `Volume` is.

Reply:
Just added this in #3675 as it's quite helpful to wrap [vtkVolume](https://vtk.org/doc/nightly/html/classvtkVolume.html). This really should be called a "vtkVolumeActor". Not sure why it's named this way

I'm of the opinion that we should accept both, as well as expanding our existing widgets to work with `pyvista.DataSet`s as well as `pyvista.Actor`s.